peerinst/templates/peerinst/collection/collection_detail.html

```diff
@@ -0,0 +1,324 @@
+                {% trans 'Correct Answers on Second Attempt' %}: {{ collection_data.correct_second_answers }}
+              </div>
+            </div>
+          </div>
+        </div>
+      </div>
+    </div>
+    <ul>
+    {% for obj in object.assignments.all %}
+    <li class="dense-list searchable">
+        <div class="mdc-card" id="card-assign">
+          <div class="mdc-typography--title bold">
+            {{ obj.title|bleach_html|safe }}
+          </div>
+        <div class="mdc-typography--caption">
+          {% trans 'Created by' %} {% for c in obj.owner.all %}
+          {{ c }}{% if not forloop.last %}, {% endif %}
+          {% endfor %}
+        </div>
+        <div class="mdc-typography--body1">
+          <div class="mdc-card" id="desciption">
+            <div class="mdc-typography--body1">
+                <b>{% trans 'Assignment Statistics' %}</b>
+            </div>
+            <div class="mdc-card__actions">
+              <div class="mdc-card__action-buttons">
+                <div class="mdc-typography--caption">
+                  <div>
+                    {% trans 'Total Answers' %}: {{ assignment_data|get_value:obj.pk|get_value:"total_answers" }}
+                  </div>
+                  <div>
+                    {% trans 'Answers Switched' %}: {{ assignment_data|get_value:obj.pk|get_value:"switches" }}
+                  </div>
+                </div>
+              </div>
+              <div class="mdc-card__action-icons">
+                <div class="mdc-card__action-buttons">
+                  <div class="mdc-typography--caption">
+                    <div>
+                      {% trans 'Correct Answers on First Attempt' %}: {{ assignment_data|get_value:obj.pk|get_value:"correct_first_answers" }}
+                    </div>
+                    <div>
+                      {% trans 'Correct Answers on Second Attempt' %}: {{ assignment_data|get_value:obj.pk|get_value:"correct_second_answers" }}
+                    </div>
+                  </div>
+                </div>
+              </div>
+            </div>
+          </div>
+          <div class="foldable">
+            <h1 class="foldable--title mdc-typography--display1 mdc-theme--secondary">
+              {{ obj.questions.count }} {% trans 'Questions' %}
+            </h1>
+            <div class="foldable--body">
+              <ul>
+                {% for question in obj.questions.all %}
+                <li class="dense-list searchable">
+                  <div class="mdc-card"
+                       id="{{ obj.id|safe }}"
+                       discipline="'{{ obj.discipline|safe }}'"
+                       category="{% for c in obj.category.all %}{{ c.title|safe }} {% endfor %}"
+                       matrix="{{ obj.get_matrix|safe }}">
+                    <div class="mdc-typography--title bold">
+                      {{ question.title|safe }}
+                    </div>
+                    <div class="mdc-typography--caption">
```

Comment:
Safe should only be put around specific user input that was bleached or json content that's checked in some way by the javascript. If django fails to render it, there's a reason for that that needs to be investigated.

Reply:
Yeah, we should update this in a few spots.  My memory is fuzzy, but a while ago we were having escape issues on valid input.  Add the bleach tag on title, text, and answer_choices (basically anything the user inputs) and remove the rest.  If you catch json in the context, I think that will still need to be marked safe (we should check the content before sending it).